libraries/rush-lib/src/logic/ProjectChangeAnalyzer.ts

```diff
@@ -292,20 +323,16 @@ export class ProjectChangeAnalyzer {
           continue;
         }
 
-        if (enableFiltering) {
-          let projectChanges: Map<string, IFileDiffStatus> | undefined = changesByProject.get(project);
-          if (!projectChanges) {
-            projectChanges = new Map();
-            changesByProject.set(project, projectChanges);
-          }
-          projectChanges.set(file, diffStatus);
-        } else {
-          changedProjects.add(project);
+        let projectChanges: Map<string, IFileDiffStatus> | undefined = changesByProject.get(project);
+        if (!projectChanges) {
+          projectChanges = new Map();
+          changesByProject.set(project, projectChanges);
         }
+        projectChanges.set(file, diffStatus);
```

Comment:
The motivation to take the filters as a parameter is that the logic here can become:
```suggestion
      const filterForProject: ((relativePath: string) => boolean) | undefined = await options.filterByProject(project);
      if (filterByProject) {
        const projectRelativePath: string = file.slice(match.index + 1);
        if (filterByProject(projectRelativePath)) {
          changedProjects.add(project);
        }
      } else {
        changedProjects.add(project);
      }
```

Reply:
You'll need to use `lookup.findLongestPrefixMatch` instead of `lookup.findChildPath`. It returns `{ value: RushConfigurationProject, index: number } | undefined`.